eclipse/sw360
# feat(VirusScanSchedulerService): Manual Scheduler service for deleting the attachment.

[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
    

- The manual scheduler service "**Delete Attachment from Local FileSystem**" will allow the user to delete the attachment which has been uploaded to local file system for virus scan which is more than 30 days older.

> * Which issue is this pull request belonging to and how is it solving it? (#1183  )
> * Did you add or update any new dependencies that are required for your change? No

Issue: #1183 

### Suggest Reviewer
> You can suggest reviewers here with an @mention.

### How To Test?
> How should these changes be tested by the reviewer?

-  Check if there is attachment stored to local file system in the directory **/opt/sw360tempattachments**.
-  If no attachment present in local FS then upload attachment and update the project,component or release that should create the attachment in local FS.
-  If the attachment which is present is more than 30 days older then it should delete that attachment by clicking the Manual scheduler service (**Attachment Deletion From Local FS**) from the Admin tab or by scheduling it through a scheduler service.
- The scheduler service user should be able to schedule it every day in the productive environment but it is also configurable.
- For scheduling:**Schedule Attachment Deletion From Local FS**, For Cancel: **Cancel Scheduled Attachment Deletion From Local FS**
- If there is some attachment which is modified today but there exists some additional attachment which is more than 30 days older in this scenario only the attachment with 30 days older should get deleted.
- Added the CVESearch service also in the manual scheduling.
- If in the directory all the attachment is more than 30 days older then it should delete the attachment along with the directory.
![image](https://user-images.githubusercontent.com/56516845/117613324-5bdde400-b184-11eb-8eb6-e5de7de371d5.png)


> Have you implemented any additional tests? No

### Checklist
Must:

Signed-off-by: ravi110336 kumar.ravindra@siemens.com
